File: backend/app/schemas/document_detail.py
"""Schémas de sortie de la page de détail d'un document."""

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import StatutDocumentEnum
from app.schemas.mouvement_bancaire import MouvementBancaireOut


class EcritureResumeOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type_ecriture: str
    numero_piece: str | None = None
    date_piece: date | None = None
    tiers: str | None = None
    montant_ht: Decimal | None = None
    taux_tva: str | None = None
    montant_tva: Decimal | None = None
    montant_ttc: Decimal
    statut_validation: str
    anomalie_detectee: bool = False
    anomalie_details: str | None = None


class DocumentDetailOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nom_fichier_original: str
    taille_octets: int | None = None
    mime_type: str | None = None
    statut: StatutDocumentEnum
    created_at: datetime

    entreprise_id: uuid.UUID | None = None
    annee: int | None = None
    mois: int | None = None
    categorie: str | None = None

    texte_ocr: str | None = None
    donnees_extraites: dict | None = None
    message_erreur: str | None = None
    type_erreur: str | None = None
    error_code: str | None = None
    saisie_topaze: bool = False

    ecriture: EcritureResumeOut | None = None
    mouvements_bancaires: list[MouvementBancaireOut] = Field(default_factory=list)

File: frontend/src/types/documentDetail.ts
import type { MouvementBancaire } from "./mouvementBancaire";

export interface EcritureResume {
  id: string;
  type_ecriture: string;
  numero_piece: string | null;
  date_piece: string | null;
  tiers: string | null;
  montant_ht: string | null;
  taux_tva: string | null;
  montant_tva: string | null;
  montant_ttc: string;
  statut_validation: string;
  anomalie_detectee: boolean;
  anomalie_details: string | null;
}

export interface DocumentDetail {
  id: string;
  nom_fichier_original: string;
  taille_octets: number | null;
  mime_type: string | null;
  statut: "en_attente" | "en_traitement" | "traite" | "valide" | "erreur";
  created_at: string;

  entreprise_id: string | null;
  annee: number | null;
  mois: number | null;
  categorie: string | null;

  texte_ocr: string | null;
  donnees_extraites: Record<string, unknown> | null;
  message_erreur: string | null;

  ecriture: EcritureResume | null;
  mouvements_bancaires?: MouvementBancaire[] | null;
}